SIGSEGV on oci push

### Operating system

Manjaro

### Kluctl Version

2.24.0-devel

### Kubernetes Version

v1.28.6

### kubectl Version

v1.29.2

### Bug description

Hi!
I'm trying to push a project to the registry (nexus/zot) but SIGSEGV occurs.

### Steps to reproduce

1) 
```
git init kluctl-test && cd kluctl-test
```
2)
```
cat <<EOF >deployment.yaml
deployments:
  - path: namespace
EOF
```
3) 
```
mkdir namespace
```
4)
```
cat <<EOF >namespace/namespace.yaml
apiVersion: v1
kind: Namespace
metadata:
  name: test-namespace
EOF
```
5) 
```
git add . && git commit -m "init"
```
6)
```
kluctl oci push --url oci://my.registry.domain/kluctl-projects
```
7) "panic: runtime error: invalid memory address or nil pointer dereference"

### Relevant log output

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x22892e8]

goroutine 1 [running]:
github.com/kluctl/kluctl/v2/cmd/kluctl/commands.(*ociPushCmd).Run(0xc000578730, {0x5368fb0, 0xc000b34e40})
	/home/runner/work/kluctl/kluctl/cmd/kluctl/commands/cmd_oci_push.go:101 +0xaa8
github.com/kluctl/kluctl/v2/cmd/kluctl/commands.(*rootCommand).buildCobraCmd.func1(0xc000afec00?, {0x292852f?, 0x4?, 0x292863f?})
	/home/runner/work/kluctl/kluctl/cmd/kluctl/commands/cobra_utils.go:73 +0x2d
github.com/spf13/cobra.(*Command).execute(0xc000afec00, {0xc0002ccd60, 0x2, 0x2})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0xabc
github.com/spf13/cobra.(*Command).ExecuteC(0xc00092c600)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/kluctl/kluctl/v2/cmd/kluctl/commands.Execute({0x5368f78, 0x6993460}, {0xc00019e010, 0x4, 0x4}, 0xc000011008)
	/home/runner/work/kluctl/kluctl/cmd/kluctl/commands/root.go:367 +0x2aa
github.com/kluctl/kluctl/v2/cmd/kluctl/commands.Main()
	/home/runner/work/kluctl/kluctl/cmd/kluctl/commands/root.go:265 +0xfe
main.main()
	/home/runner/work/kluctl/kluctl/cmd/main.go:27 +0x47
```

--- a/cmd/kluctl/commands/cmd_oci_push.go
+++ b/cmd/kluctl/commands/cmd_oci_push.go
@@ -98,9 +98,11 @@ func (cmd *ociPushCmd) Run(ctx context.Context) error {
 	}
 
 	meta := client.Metadata{
-		Source:      gitInfo.Url.String(),
 		Revision:    gitInfo.Commit,
 		Annotations: annotations,
+	}
+	if gitInfo.Url != nil {
+		meta.Source = gitInfo.Url.String()
 	}
 
 	ctx, cancel := context.WithTimeout(ctx, cmd.Timeout)

